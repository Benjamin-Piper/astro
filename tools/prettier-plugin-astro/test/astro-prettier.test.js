import { format } from './test-utils.js';
import { promises as fs } from 'fs';
import { fileURLToPath } from 'url';

const readFile = (path) => fs.readFile(fileURLToPath(new URL(`./fixtures/${path}`, import.meta.url)), 'utf8').then((res) => res.replace(/\r\n/g, '\n'));

describe('Prettier formatting', () => {
  test('can format a basic Astro file', async () => {
    const src = await readFile('basic.astro');
    expect(format(src)).toMatchSnapshot();
  });

  test('can format an Astro file with frontmatter', async () => {
    const src = await readFile('frontmatter.astro');
    expect(format(src)).toMatchSnapshot();
  });

  test('can format an Astro file with embedded JSX expressions', async () => {
    const src = await readFile('embedded-expr.astro');
    expect(format(src)).toMatchSnapshot();
  });
});
