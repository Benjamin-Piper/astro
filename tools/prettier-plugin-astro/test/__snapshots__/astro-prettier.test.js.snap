// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prettier formatting can format a basic Astro file 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Document</title>
  </head>
  <body>
    <h1>Hello world!</h1>
  </body>
</html>
"
`;

exports[`Prettier formatting can format an Astro file with embedded JSX expressions 1`] = `
"---
import Color from \\"../components/Color.jsx\\";

let title = \\"My Site\\";

const colors = [\\"red\\", \\"yellow\\", \\"blue\\"];
---

<html lang=\\"en\\">
  <head>
    <title>My site</title>
  </head>
  <body>
    <h1>{title}</h1>

    <script $ lang=\\"ts\\">
      \\"I'm some super long text and oh boy I sure do hope this formatter doesn't break me!\\";
    </script>

    <script $ lang=\\"ts\\">
      colors.map((color) => (
        <div>
          <Color name={color} />
        </div>
      ));
    </script>
  </body>
</html>
"
`;

exports[`Prettier formatting can format an Astro file with frontmatter 1`] = `
"---
import Color from \\"../components/Color.jsx\\";

let title = \\"My Site\\";

const colors = [\\"red\\", \\"yellow\\", \\"blue\\"];
---

<html lang=\\"en\\">
  <head>
    <title>My site</title>
  </head>
  <body>
    <h1>{title}</h1>
  </body>
</html>
"
`;
